CIP-0144/README.md

```diff
@@ -0,0 +1,356 @@
+
+### CPS-10
+
+[CPS-0010](https://github.com/Ryun1/CIPs/blob/cps-wallet-connector/CPS-0010/README.md) discusses several limitations of CIP-30. In this CIP we try to solve the issues pointed out in that CPS. There are still some areas to improve on: things like the event listener API are intentionally left out of this CIP to prevent bloat of scope. We welcome future CIPs, or updates to this CIP to refine any limitation that is not tackled in this document.
+
+### Use of CBOR representations
+
+In addition to the points described in CPS-10, the CIP-30 standard uses CBOR encoding for all data passed from the wallet, e.g. addresses and UTxOs. Interpreting this data within the app requires CBOR decoding functionality that is tedious to implement manually, and so users resort to using cardano-serialization-lib or its close alternative, cardano-multiplatform-lib, which both require loading a WebAssembly blob of >1M in size.
+
+For comparison, to start a new Web3 app on Ethereum there is no need to use a library for data serialization. It’s possible to interact with a provider object that is given by the wallet directly, although there are libraries to further simplify this. Using CBOR looks unnecessary for most dApps, given that JSON is a de-facto standard for web data serialization.
+
+In this work we build on the work carried out in [CIP-0116](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0116) defining a universal JSON encoding for Cardano domain types.
+
+## Specification
+
+The goal of this CIP is to provide a better alternative to CIP-30, providing a base on which we can define a CIP-30 compatible API, and further extensions in the future. Specifically we make the following contributions:
+
+- A clear separation between the connection mechanism and the different APIs offered by the wallet.
+- Using JSON for Cardano domain types instead of CBOR
+- Add versioning support to the extension API
+- Define the API in a transport agnostic way
+
+### Transport agnostic operations
+
+In its current state, CIP-30 defines it API through a specific transport layer, namely an injected Javascript object.
+In this CIP we want to be able to define an API without committing to a specific transport layer. Implementors of this API can choose to support this API through several transports such as: HTTP, an injected Javascript object, JSON-RPC etc.
+Furthermore, we want to use JSON-schema to clearly define the types that each method, or operation, expects to receive or returns.
+To keep the specification abstract we will use the word "operation" to describe the actions supported by the API: these would map to "endpoints" for an HTTP implementation of the API, or to "methods" for an implementation based on an injected Javascript object.
+
+
+We will use the following schema to define operations:
+
+```json
+{
```

Comment:
```suggestion
{
  "connectionVersion": { "type": "semVar" },
```

does it make sense to have the connection API versioned?

Reply:
I think so, I added an endpoint both to the connection API and to the different extensions to report their `apiVersion`. Adding it here however, would potentially make each endpoint have it's own separate version which might be a bit too much complexity